javaslang/src/main/java/javaslang/collection/BitSet.java

```diff
@@ -0,0 +1,1024 @@
+                    .filter(i -> (getWord(i >> ADDRESS_BITS_PER_WORD) & (1L << i)) != 0)
+                    .map(fromInt)
+                    .iterator();
+        }
+
+        @Override
+        public int length() {
+            int len = 0;
+            for (int i = 0; i < getWordsNum(); i++) {
+                len += Long.bitCount(getWord(i));
+            }
+            return len;
+        }
+
+        @Override
+        public BitSet<T> take(long n) {
+            if (n <= 0) {
+                return createEmpty();
+            } else if (n >= length()) {
+                return this;
+            } else {
+                return createFromAll(iterator().take(n));
+            }
+        }
+
+        @Override
+        public BitSet<T> takeRight(long n) {
+            if (n <= 0) {
+                return createEmpty();
+            } else if (n >= length()) {
+                return this;
+            } else {
+                return createFromAll(iterator().takeRight(n));
+            }
+        }
+
+        @Override
+        public BitSet<T> remove(T t) {
+            if (contains(t)) {
+                final int element = toInt.apply(t);
+                final long[] copy = copyExpand(getWordsNum());
+                unsetElement(copy, element);
+                return fromBitMaskNoCopy(shrink(copy));
+            } else {
+                return this;
+            }
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public BitSet<T> removeAll(Iterable<? extends T> elements) {
+            final Stream<Integer> source = Stream.ofAll(elements).map(toInt);
+            final long[] copy = copyExpand(getWordsNum());
+            source.forEach(element -> {
+                unsetElement(copy, element);
+            });
+            return fromBitMaskNoCopy(shrink(copy));
+        }
+
+        @Override
+        public String toString() {
+            return mkString(stringPrefix() + "(", ", ", ")");
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (o == this) {
+                return true;
+            } else if (o instanceof BitSet) {
+                final BitSet<?> that = (BitSet<?>) o;
+                return Collections.equals(this, that);
+            } else {
+                return false;
+            }
+        }
+
+        @Override
+        public int hashCode() {
+            return Collections.hash(this);
+        }
+    }
+
+    class BitSet1<T> extends AbstractBitSet<T> {
+
+        private static final long serialVersionUID = 1L;
+
+        private final long elements;
+
+        BitSet1(Function1<Integer, T> fromInt, Function1<T, Integer> toInt, long elements) {
+            super(fromInt, toInt);
+            this.elements = elements;
+        }
+
+        @Override
+        int getWordsNum() {
+            return 1;
+        }
+
+        @Override
+        long[] copyExpand(int wordsNum) {
+            if(wordsNum < 1) {
```

Comment:
Not sure if it's more readable or not, just an alternative one-liner instead of 899-902:
```java
long[] arr = new long[Math.max(1, wordsNum)];
```